src/pyscipopt/scip.pxi

```diff
@@ -5439,6 +5533,76 @@ cdef class Model:
         """Get an estimation of the final tree size """
         return SCIPgetTreesizeEstimation(self._scip)
 
+@dataclass
+class Statistics:
+    # Total time since model was created
+    total_time: float
+    # Time spent solving the problem
+    solving_time: float
+    # Time spent on presolving
+    presolving_time: float
+    # Time spent on reading
+    reading_time: float
+    # Time spent on copying
+    copying_time: float
+    # Name of problem
+    problem_name: str
+    # Name of presolved problem
+    presolved_problem_name: str
+    # Dictionary with number of variables by type 
+    _variables: dict
+    # Dictionary with number of presolved variables by type 
+    _presolved_variables: dict
+    # Dictionary with number of constraints by type
+    _constraints: dict
+    # Dictionary with number of presolved constraints by type
+    _presolved_constraints: dict
+    # The number of restarts it took to solve the problem (TODO: check this)
+    n_runs: int
+    # The number of nodes explored in the branch-and-bound tree
+    n_nodes: int
+    # number of found solutions
+    solutions_found: int
+    # objective value of first found solution
+    first_solution: float    
+    # The best primal bound found 
+    primal_bound: float
+    # The best dual bound found
+    dual_bound: float
+    # The gap between the primal and dual bounds
+    gap: float
+    # The primal-dual integral 
+    primal_dual_integral: float
+
+    def _populate_remaining(self):
+        # number of variables in the model
+        self.n_vars: int = self._variables["total"]
```

Comment:
These should be computed properties exposed as methods, would make the values more visible also easier to add documentation later.

Reply:
Done!